internal/controllers/aggregation/status.go

```diff
@@ -0,0 +1,85 @@
+package aggregation
+
+import (
+	"context"
+	"fmt"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/Azure/eno/internal/manager"
+	"github.com/go-logr/logr"
+)
+
+type statusController struct {
+	client client.Client
+}
+
+func NewStatusController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Composition{}).
+		Owns(&apiv1.ResourceSlice{}).
+		WithLogConstructor(manager.NewLogConstructor(mgr, "statusAggregationController")).
+		Complete(&statusController{
+			client: mgr.GetClient(),
+		})
+}
+
+func (s *statusController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+
+	comp := &apiv1.Composition{}
+	err := s.client.Get(ctx, req.NamespacedName, comp)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting composition: %w", err))
+	}
+	if comp.Status.CurrentState == nil || !comp.Status.CurrentState.Synthesized || (comp.Status.CurrentState.Ready && comp.Status.CurrentState.Reconciled) {
+		return ctrl.Result{}, nil
+	}
+
+	ready := true
+	reconciled := true
+	for _, ref := range comp.Status.CurrentState.ResourceSlices {
+		slice := &apiv1.ResourceSlice{}
+		slice.Name = ref.Name
+		slice.Namespace = comp.Namespace
+		err := s.client.Get(ctx, client.ObjectKeyFromObject(slice), slice)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("getting resource slice: %w", err)
+		}
+
+		// Status might be lagging behind
+		if len(slice.Status.Resources) == 0 && len(slice.Spec.Resources) > 0 {
+			ready = false
+			reconciled = false
+			break
+		}
+
+		for _, state := range slice.Status.Resources {
```

Comment:
Why && with `reconciled` and `ready`? You're just avoiding setting the variables to false repeatedly, but it shouldn't make a difference, right? Maybe I'm missing something

Reply:
You're completely right - this was just some leftovers from a slightly different implementation